### PR TITLE
docs(recipes): Add Netlify section in Deploying

### LINKED ISSUE #15535: [docs][recipes] deploy to Netlify
## Summary

As part of the recipes initiative and the [Deploying to Netlify workflow](#13804), there should be a recipe on Deploying to Netlify. Recipes have a new [actionable format](https://www.gatsbyjs.org/contributing/docs-templates/#recipes) where the steps are included inline. Because a lot of this workflow happens outside of Gatsby itself, this recipe may take some experimentation to find the right amount of content. This recipe should leverage Prerequisites to list the required setup steps without having to show them all inline, and focus on the Gatsby-specific parts.

Recipes now follow this format:

- Requirements or prerequisites 
    - Gatsby CLI
    - A Gatsby site
- Step-by-step instructions
- Relevant links to any relevant information
    - More on [preparing a site for deployment](https://www.gatsbyjs.org/docs/preparing-for-deployment/)

This recipe should go into the "Deploying" section of [`docs/docs/recipes.md`](https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/recipes.md) after the [Preparing for deployment recipe](#15319), and replacing the existing text. Docs URLs to gatsbyjs.org should be relative. There is a recipe template in the contributing docs: https://www.gatsbyjs.org/contributing/docs-templates/#recipes

For reference, the new recipe collection is also listed in a spreadsheet: https://docs.google.com/spreadsheets/d/1yNUQi_-3TsuDtE98hCIJQvLZ6X1YHlDmC-EJwOGxiHk/edit#gid=1252007048

### PR DESCRIPTION
**Pull request by luizcieslak**
## Description

This adds Deploy to Netlify section in Deploying category, using `netlify-cli` to build and relating Hosting on Netlify documentation and `gatsby-plugin-netlify` as additional resources.

## Related Issues

Fixes #15535

### CODE REVIEW

File: docs/docs/recipes.md [hunk 1]
```diff
@@ -129,6 +129,38 @@ Showtime.
 - Learn about [performance optimization](/docs/performance/)
 - Read about [other deployment related topics](/docs/deploying-and-hosting/)
 
+
+### Deploying to Netlify
+
+Use `netlify-cli` to deploy your gatsby application without getting out of command line interface.
```
Comment (cglong) [inline docs/docs/recipes.md]: A few grammatical suggestions 🙂
```suggestion
Use [`netlify-cli`](https://www.netlify.com/docs/cli/) to deploy your Gatsby application without leaving the command line interface.
```

Comment (luizcieslak) [reply docs/docs/recipes.md]: Thanks a lot!

**Comment by luizcieslak** [comment]
@marcysutton Should I run prettier in these markdown files? It did't pass the `lint` step in CircleCI

File: docs/docs/recipes.md [hunk 2]
```diff
@@ -129,6 +129,37 @@ Showtime.
...
 - Read about [other deployment related topics](/docs/deploying-and-hosting/)
 
+### Deploying to Netlify
+
+Use [`netlify-cli`](https://www.netlify.com/docs/cli/) to deploy your Gatsby application without leaving the command line interface.
+
+#### Prerequisites
+
+- A Gatsby site with a single component `index.js`.
+- [netlify-cli](https://www.npmjs.com/package/netlify-cli) npm package
```
[the remark this answers is not in the document]
Comment (luizcieslak) [reply docs/docs/recipes.md]: When you run `netlify login` a new browser is opened to login and authorize "Netlify CLI" application or to create a new account if its needed.

Do you think that a Netlify account is a prerequisite?

**Pull request merged**